docs/docs/reference-react.md

```diff
@@ -1,24 +1,67 @@
 ---
-id: reference-react
+id: react-api
 title: React Top-Level API
 layout: docs
 category: Reference
-permalink: docs/reference-react.html
+permalink: docs/react-api.html
 redirect_from: "/docs/reference.html"
 ---
 
 `React` is the entry point to the React library.
 
-## Reference
+## Overview
+
+### Components
+
+React components let you split the UI into independent, reusable pieces, and think about each piece in isolation. React components can be defined by subclassing `React.Component`.
 
  - [`React.Component`](#react.component)
- - [`createClass()`](#createclass)
- - [`createElement()`](#createlement)
- - [`cloneElement()`](#cloneelement)
+ - [`createClass()`](#react.createclass)
+
+#### Typechecking with PropTypes
+
+You can use `React.PropTypes` to run typechecking on the props for a component.
+
+ - [`React.PropTypes`](#react.proptypes)
+ - [`React.PropTypes.array`](#react.proptypes.array)
+ - [`React.PropTypes.bool`](#react.proptypes.bool)
+ - [`React.PropTypes.func`](#react.proptypes.func)
+ - [`React.PropTypes.number`](#react.proptypes.number)
+ - [`React.PropTypes.object`](#react.proptypes.object)
+ - [`React.PropTypes.string`](#react.proptypes.string)
+ - [`React.PropTypes.symbol`](#react.proptypes.symbol)
+ - [`React.PropTypes.node`](#react.proptypes.node)
+ - [`React.PropTypes.element`](#react.proptypes.element)
+ - [`React.PropTypes.instanceOf()`](#react.proptypes.instanceof)
+ - [`React.PropTypes.oneOf()`](#react.proptypes.oneof)
+ - [`React.PropTypes.oneOfType()`](#react.proptypes.oneoftype)
+ - [`React.PropTypes.arrayOf()`](#react.proptypes.arrayof)
+ - [`React.PropTypes.objectOf()`](#react.proptypes.objectof)
+ - [`React.PropTypes.shape()`](#react.proptypes.shape)
+ - [`React.PropTypes.any`](#react.proptypes.any)
+
+Validators treat props as optional by default. You can use `isRequired` to make sure a warning is shown if the prop is not provided.
+
+ - [`isRequired`](#isrequired)
+
+### Creating React Elements
+
+We recommend [using JSX](/react/docs/introducing-jsx.html) to describe what your UI should look like. Each JSX element is just syntactic sugar for calling [`React.createElement()`](#createelement). You will not typically invoke the following methods directly if you are using JSX.
+
+ - [`createElement()`](#createelement)
  - [`createFactory()`](#createfactory)
+
+See [Using React without JSX](/react/docs/react-without-jsx.html) for more information.
+
+### Other APIs
```

Comment:
Can this be "Transforming Elements" or something? Both these APIs deal with elements and children.